FogFog-iOS/FogFog-iOS/Presentation/Map/MapViewController.swift

```diff
@@ -48,16 +60,52 @@ final class MapViewController: BaseViewController {
         setLocation()
     }
     
+    // MARK: UI
+    override func setStyle() {
+        let blurEffect = UIBlurEffect(style: .dark)
+        blurEffectView = UIVisualEffectView(effect: blurEffect)
+        blurEffectView.alpha = 0.6
+        blurEffectView.isHidden = true
+    }
+    
     override func setLayout() {
         
-        view.addSubview(navigationView)
+        view.addSubviews([navigationView, blurEffectView, sideBarView])
         
         navigationView.snp.makeConstraints {
             $0.top.equalToSuperview()
             $0.height.equalTo(102)
             $0.leading.trailing.equalToSuperview()
         }
+        
+        blurEffectView.snp.makeConstraints {
+            $0.top.bottom.leading.trailing.equalToSuperview()
+        }
+        
+        sideBarView.snp.makeConstraints {
+            $0.top.bottom.equalToSuperview()
+            $0.trailing.equalToSuperview().inset(-309)
+            $0.width.equalTo(309)
+        }
+    }
+}
+
+// MARK: - Bind
+extension MapViewController {
+    
+    private func bind() {
+        
+        output.isVisible
+            .asDriver()
+            .drive { state in
+                self.setSideBarViewLayout(isVisible: state)
```

Comment:
p1;
순환 참조가 발생할 수 있을 것 같아요!

```suggestion
        output.isVisible
            .asDriver()
            .drive { [weak self] state in
                self?.setSideBarViewLayout(isVisible: state)
```

아니면 RxSwift6 부터 지원하는 `.withUnretained(self)` 한번 찾아봐도 좋을 것 같습니다 :)